### PR TITLE
feat(pod-delete): Improving banner name for pod-delete experiment

### LINKED ISSUE #1235: Better Consistent Chaos Runner Logs
While running lots of experiments in a row, I'm watching the logs for each experiment's chaos running container while they run and find it hard to see what settings experiments are being run with to confirm it's following my YAML file settings (and any optional variables I didn't specify in the YAML, seeing what defaults have been set), seeing if the RAMP_UP time is being respected, and more detail on exactly what the experiment is doing.

Without looking at the framework code of the runner, I would put some consistent logging around an experiment so every experiment outputs a consistent startup and shutdown so its easy to see what was going on/settings - and request clearer logging for each experiment to understand what's going on during the running, which will be different depending on the experiment.

For startup I'd like to see:

1. Experiment name and all the variables its using with their values
2. If RAMP_TIME is set, something in the logs to indicate RAMP_TIME was set and the experiment is waiting for X seconds till it runs so I can confirm there definitely was a delay between setup and running the test
3. Some clear markers in the log to seperate setup/waiting/running/shutdown - I saw most have "Brace yourself, experiment begins!" in them, markers like this make the logs much clearer so add some more

For teardown:

1. A clear indication of overall experiment result, it shows this line but I'm not sure what all the numbers mean: : ok=36   changed=21   unreachable=0    failed=0
2. Any RAMP_TIME at end of test clearly marked out in logs

Seeing all the experiment variables and RAMP_TIME is most important, as in some experiments I'm setting things and don't know for sure if they're being set or used properly so having some logs for debugging would help!

### PR DESCRIPTION
**Pull request by ispeakc0de**
Signed-off-by: shubhamchaudhary <shubham.chaudhary@mayadata.io>

**What this PR does / why we need it**:

- Improving tasks name

- Adding prereq, prepare, inject, status, ramp, event sections in experiment 

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #1235

**Checklist**

* [ ] Does this PR have a corresponding GitHub issue?
* [ ] Have you included relevant README for the chaoslib/experiment with details?
* [ ] Have you added debug messages where necessary? 
* [ ] Have you added task comments where necessary? 
* [ ] Have you tested the changes for possible failure conditions?
* [ ] Have you provided the positive & negative test logs for the litmusbook execution?
* [ ] Does the litmusbook ensure idempotency of cluster state?, i.e., is cluster restored to original state?
* [ ] Have you used non-shell/command modules for Kubernetes tasks?
* [ ] Have you (jinja) templatized custom scripts that is run by the litmusbook, if any? 
* [ ] Have you (jinja) templatized Kubernetes deployment manifests used by the litmusbook, if any?
* [ ] Have you reused/created util functions instead of repeating tasks in the litmusbook?
* [ ] Do the artifacts follow the appropriate directory structure? 
* [ ] Have you isolated storage (eg: OpenEBS) specific implementations, checks? 
* [ ] Have you isolated platform (eg: baremetal kubeadm/openshift/aws/gcloud) specific implementations, checks?  
* [ ] Are the ansible facts well defined? Is the scope explicitly set for playbook & included utils?
* [ ] Have you ensured minimum/careful usage of shell utilities (awk, grep, sed, cut, xargs etc.,)?
* [ ] Can the limtusbook be executed both from within & outside a container (configurable paths, no hardcode)?
* [ ] Can you suggest the minimal resource requirements for the litmusbook execution?
* [ ] Does the litmusbook job artifact carry comments/default options/range for the ENV tunables?
* [ ] Has the litmusbooks been linted? 

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by ispeakc0de** [comment]
**Sample Logs**
```

2020-03-11 18:45:54.740632 ************ BRACE YOURSELF, EXPERIMENT BEGINS! ************ 

2020-03-11 18:46:02.133568 Step: Gathering Facts 

2020-03-11 18:46:03.350398 Step: [PreReq] Identify the chaos util for pod-delete experiment 

2020-03-11 18:46:03.496717 Step: [PreReq]: Including the chaos util for pod-delete experiment 

2020-03-11 18:46:03.658880 Step: [PreReq]: Constructing the chaos result name 
included: /utils/runtime/update_chaos_result_resource.yml for 127.0.0.1

2020-03-11 18:46:04.470061 Step: [PreReq]: Generate the chaos result CR to reflect SOT (Start of Test) 

2020-03-11 18:46:06.388901 Step: PreReq]: Apply the chaos result CR for pod-delete experiment 
included: /utils/common/status_app_pod.yml for 127.0.0.1

2020-03-11 18:46:07.930033 Step: [Status]: Checking whether application pods are in running state 

2020-03-11 18:46:09.146255 Step: [Status]: Checking whether application containers are in running state 
included: /utils/common/generate-kubernetes-chaos-events.yml for 127.0.0.1

2020-03-11 18:46:10.323068 Step: [Event]: Obtain the current date in ISO8601 format 

2020-03-11 18:46:10.476231 Step: set_fact 

2020-03-11 18:46:12.465460 Step: [Event]: Creating a PreChaosCheck event in the chaosengine 
included: /chaoslib/litmus/pod_failure_by_litmus.yml for 127.0.0.1

2020-03-11 18:46:12.856337 Step: [Prepare]: Deriving the chaos iterations 

2020-03-11 18:46:13.196258 Step: debug 
details => {
    "msg": "***** Waiting for the ramp interval of 20s *****"
}

2020-03-11 18:46:33.987283 Step: [Ramp]: Wait for the specified ramp time before injecting chaos 
included: /chaoslib/litmus/kill_random_pod.yml for 127.0.0.1 => (item=1)
included: /chaoslib/litmus/kill_random_pod.yml for 127.0.0.1 => (item=2)
included: /chaoslib/litmus/kill_random_pod.yml for 127.0.0.1 => (item=3)

2020-03-11 18:46:36.292491 Step: [Prepare]: Fetching list of all pods with run=nginx label from shubham namespace 

2020-03-11 18:46:36.438831 Step: [Prepare]: Creating a empty list to storing pod for deletion 

2020-03-11 18:46:36.603278 Step: [Prepare]: Select a random pod to kill 

2020-03-11 18:46:36.774445 Step: [Prepare]: Adding nginx1-88b9989dd-jt56q pod in list for deletion 

2020-03-11 18:46:36.937005 Step: set_fact 
included: /utils/common/generate-kubernetes-chaos-events.yml for 127.0.0.1

2020-03-11 18:46:38.505280 Step: [Event]: Obtain the current date in ISO8601 format 

2020-03-11 18:46:38.674904 Step: set_fact 

2020-03-11 18:46:40.402203 Step: [Event]: Creating a ChaosInject event in the chaosengine 

2020-03-11 18:46:40.546868 Step: debug 
details => {
    "msg": "Killing nginx1-88b9989dd-jt56q pod"
}

2020-03-11 18:46:41.841982 Step: [Inject]: Forcefully killing the application pod 
Pausing for 10 seconds
(ctrl+C then 'C' = continue early, ctrl+C then 'A' = abort)

2020-03-11 18:46:52.163708 Step: [Wait]: Wait for the chaos interval 10s 
included: /utils/common/status_app_pod.yml for 127.0.0.1

2020-03-11 18:46:53.506576 Step: [Status]: Checking whether application pods are in running state 

2020-03-11 18:46:54.690994 Step: [Status]: Checking whether application containers are in running state 

2020-03-11 18:46:56.245833 Step: [Prepare]: Fetching list of all pods with run=nginx label from shubham namespace 

2020-03-11 18:46:56.404240 Step: [Prepare]: Creating a empty list to storing pod for deletion 

2020-03-11 18:46:56.569268 Step: [Prepare]: Select a random pod to kill 

2020-03-11 18:46:56.737519 Step: [Prepare]: Adding nginx-88b9989dd-2rbpm pod in list for deletion 

2020-03-11 18:46:56.893587 Step: set_fact 
included: /utils/common/generate-kubernetes-chaos-events.yml for 127.0.0.1

2020-03-11 18:46:58.456031 Step: [Event]: Obtain the current date in ISO8601 format 

2020-03-11 18:46:58.609319 Step: set_fact 

2020-03-11 18:47:00.121964 Step: [Event]: Creating a ChaosInject event in the chaosengine 

2020-03-11 18:47:00.269872 Step: debug 
details => {
    "msg": "Killing nginx-88b9989dd-2rbpm pod"
}

2020-03-11 18:47:01.598780 Step: [Inject]: Forcefully killing the application pod 
Pausing for 10 seconds
(ctrl+C then 'C' = continue early, ctrl+C then 'A' = abort)

2020-03-11 18:47:11.858727 Step: [Wait]: Wait for the chaos interval 10s 
included: /utils/common/status_app_pod.yml for 127.0.0.1

2020-03-11 18:47:13.212025 Step: [Status]: Checking whether application pods are in running state 

2020-03-11 18:47:14.446337 Step: [Status]: Checking whether application containers are in running state 

2020-03-11 18:47:16.083905 Step: [Prepare]: Fetching list of all pods with run=nginx label from shubham namespace 

2020-03-11 18:47:16.227686 Step: [Prepare]: Creating a empty list to storing pod for deletion 

2020-03-11 18:47:16.387150 Step: [Prepare]: Select a random pod to kill 

2020-03-11 18:47:16.552548 Step: [Prepare]: Adding nginx-88b9989dd-cp5rx pod in list for deletion 

2020-03-11 18:47:16.718879 Step: set_fact 
included: /utils/common/generate-kubernetes-chaos-events.yml for 127.0.0.1

2020-03-11 18:47:18.247237 Step: [Event]: Obtain the current date in ISO8601 format 

2020-03-11 18:47:18.460565 Step: set_fact 

2020-03-11 18:47:19.989190 Step: [Event]: Creating a ChaosInject event in the chaosengine 

2020-03-11 18:47:20.130580 Step: debug 
details => {
    "msg": "Killing nginx-88b9989dd-cp5rx pod"
}

2020-03-11 18:47:21.379965 Step: [Inject]: Forcefully killing the application pod 
Pausing for 10 seconds
(ctrl+C then 'C' = continue early, ctrl+C then 'A' = abort)

2020-03-11 18:47:31.643965 Step: [Wait]: Wait for the chaos interval 10s 
included: /utils/common/status_app_pod.yml for 127.0.0.1

2020-03-11 18:47:33.013028 Step: [Status]: Checking whether application pods are in running state 

2020-03-11 18:47:34.238647 Step: [Status]: Checking whether application containers are in running state 

2020-03-11 18:47:34.390925 Step: debug 
details => {
    "msg": "***** Waiting for the ramp interval of 20s *****"
}

2020-03-11 18:47:54.801905 Step: [Ramp]: Wait for the specified ramp time after injecting chaos 
included: /utils/common/status_app_pod.yml for 127.0.0.1

2020-03-11 18:47:56.179060 Step: [Status]: Checking whether application pods are in running state 

2020-03-11 18:47:57.395685 Step: [Status]: Checking whether application containers are in running state 
included: /utils/common/generate-kubernetes-chaos-events.yml for 127.0.0.1

2020-03-11 18:47:58.549126 Step: [Event]: Obtain the current date in ISO8601 format 

2020-03-11 18:47:58.691568 Step: set_fact 

2020-03-11 18:48:00.205812 Step: [Event]: Creating a PostChaosCheck event in the chaosengine 

2020-03-11 18:48:00.351772 Step: set_fact 

2020-03-11 18:48:00.502540 Step: [Result]: Getting the final result of pod-delete experiment 
details => {
    "msg": "pod-delete experiment has been Passed"
}
included: /utils/runtime/getting_failure_step.yml for 127.0.0.1

2020-03-11 18:48:01.212030 Step: [Failure-Detection]: Recording the failed step name 
included: /utils/common/generate-kubernetes-chaos-events.yml for 127.0.0.1

2020-03-11 18:48:02.384650 Step: [Event]: Obtain the current date in ISO8601 format 

2020-03-11 18:48:02.558064 Step: set_fact 

2020-03-11 18:48:04.107174 Step: [Event]: Creating a Summary event in the chaosengine 
included: /utils/runtime/update_chaos_result_resource.yml for 127.0.0.1

2020-03-11 18:48:05.719909 Step: [The END]: Update the chaos result CR to reflect EOT (End of Test) 

2020-03-11 18:48:07.208920 Step: [The END]: Apply the chaos result CR for pod-delete experiment 

2020-03-11 18:48:07.213233 ************ RELAX, EXPERIMENT ENDS! ************ 
127.0.0.1                  : ok=86   changed=30   unreachable=0    failed=0 

```

**Comment by dgildeh** [comment]
Yes this looks much better!

File: chaoslib/litmus/kill_random_pod.yml [hunk 1]
```diff
@@ -10,17 +10,17 @@
           - status.phase=Running
       register: pod_list
 
-    - name: Initialize deletion list
+    - name: "[Prepare]: Creating a empty list to storing pod for deletion"
```
Comment (ksatchit) [inline chaoslib/litmus/kill_random_pod.yml]:
```suggestion
    - name: "[Prepare]: Trying to the target application pod.."
```

File: chaoslib/litmus/kill_random_pod.yml [hunk 2]
```diff
@@ -10,17 +10,17 @@
...
 
     - block: 
 
-        - name: Select a random pod to kill
+        - name: "[Prepare]: Select a random pod to kill"
           set_fact:
             app_pod_random: "{{ pod_list.resources | random | json_query('metadata.name') }}"
 
-        - name: Construct the deletion list with single random pod 
+        - name: "[Prepare]: Adding {{ app_pod_random }} pod in list for deletion"
```
Comment (ksatchit) [inline chaoslib/litmus/kill_random_pod.yml]:
```suggestion
        - name: "[Prepare]: Adding {{ app_pod_random }} pod as a chaos candidate"
```

**Pull request merged**